content/docs/user-guide/components/reference/atom/debug-rendering.md

```diff
@@ -0,0 +1,91 @@
+| **Debug Light Intensity** | The intensity of the debug directional light. | 2.0 |
+
+
+## Material Override Properties
+
+The Debug Render Component allows users to override material values for all materials in the scene. This gives users extra options which which to vizualize and debug lighting and materials in the scene. For example, one use case would be to override the color of all objects in the scene to white or grey in order to get a better sense of the lighting in the scene.
+
+Values that can be overriden are Base Color, Roughness and Metallic. Normals maps and detail normal maps cannot be overriden but they can be disabled (if both are disabled then materials in the scene will use vertex normals only).
+
+| Property | Description | Default Value |
+|-|-|-|
+| **Override Base Color** | Whether to override the base color for all materials in the scene to the value specified below. | Off |
+| **Base Color Value** | The value used to override base color on all materials in the scene. For example you can set this to red to see what your scene would look like if it were entirely red. | RGB (128, 128, 128) |
+| **Override Roughness** | Whether to override the roughness for all materials in the scene to the value specified below. | Off |
+| **Roughness Value** | The value used to override roughness on all materials in the scene. For example you can set this to zero if you want to visualize a very glossy and reflective scene. | 1.0 |
+| **Override Metallic** | Whether to override the metallic for all materials in the scene to the value specified below. | Off |
+| **Metallic Value** | The value used to override metallic on all materials in the scene. For example you can set this one to make all materials in your scene metals. | 0.0 |
+| **Enable Normal Maps** | Whether to enable/disable normal maps on all materials in the scene (note that disabling normal maps will also disable detail normal maps, and thus only the vertex normal will be used in shading calculations). | On |
+| **Enable Detail Normal Maps** | Whether to enable/disable detail normal maps on all materials in the scene (note that not all materials provide or support detail normal maps, therefore those materials will be unaffected by this option). | On |
+
+
+## Custom Debug Properties
+
+When writing shaders, it can be useful to have immediate access to a tweakable value for debug purposes. The Debug Render Component provides access to **four debug booleans** and **four debug floats** that are passed into the Scene Shader Resource Group. These values are accessible from any shader that includes the Scene SRG.
+
+**NOTE**: It is imperative that you delete any usage of these variables from your shaders when submitting your code. These values are strictly to facilitate local debugging and aren't meant for any other use. If you fail to delete your usage of these variable before you submit, another shader author might experience unwanted side effects when using these values.
```

Comment:
```suggestion
{{< important >}}
When submitting your shader code, you must delete any usage of these variables from your shaders. These values are strictly used to facilitate local debugging and aren't meant for any other use. If you don't delete these variable in your shader code before you submit, another shader author may experience unwanted side effects when using these values.
{{< /important >}}
```

Reply:
This callout is directed at contributors assuming they are writing shaders for the O3DE project. Seems a bit out of place here as this topic is about using this debug rendering component, not shader authoring. Should this be moved elsewhere in the docs, perhaps to the SRG docs above? If not, can it be rephrased or clarified for developers who are working on game projects? "When contributing your shader code", what if they are working locally and not contributing to any project? Does this advice apply to runtime shaders in general?

Reply:
These aren't variable you would normally use while authoring shaders, they would only be used in tandem with the debug component, so this is a good place. As for game projects, this advice still applies, it is best never to check in usage of these variables as that may then conflict with someone else's usage (similar to how you wouldn't use printf debugging and then check that in after you've finished debugging). It's purely there for convenience so the user doesn't have to do the plumbing themselves if they want to modify shader variables at runtime.

Reply:
Given the context above, I've revised this to the following: 
> These variables are only meant to help you debug your shaders locally. For best practice, when you are done debugging, we recommend that you delete any usage of these variables. It's especially important to delete them if you are contributing shader code either to the source repo or your team's own repo. If you don't, another shader author may experience unwanted side effects when using the same variables.